File: 0566-reshape-the-matrix/0566-reshape-the-matrix.cpp
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& mat, int r, int c) {
        vector<int> arr;
        if((r*c) != mat.size() * mat[0].size() ){
            return mat;
        }

        for(int i=0; i<mat.size(); i++){
            for(int j=0; j<mat[i].size(); j++){
                arr.push_back(mat[i][j]);
            }
        }

        vector<vector<int>> ans(r,vector<int>(c));

        int k=0;
        for(int i=0; i<r; i++){
            for(int j=0; j<c; j++){
                ans[i][j]=arr[k++];
            }
        }
        return ans;
    }
};

File: 0566-reshape-the-matrix/0566-reshape-the-matrix.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        arr=[]
        ans=[]
        for i in mat:
            for j in i:
                arr.append(j)
        n=0
        if(len(arr) != r*c):
            return mat
            
        for i in range(r):
            tl=[]
            for j in range(c):
                tl.append(arr[n])
                n+=1
            ans.append(tl)
        return ans


        